src/Asm89/Stack/CorsService.php

```diff
@@ -114,22 +129,40 @@ private function buildPreflightCheckResponse(Request $request)
             $response->headers->set('Access-Control-Allow-Credentials', 'true');
         }
 
-        $response->headers->set('Access-Control-Allow-Origin', $request->headers->get('Origin'));
+        $this->addAllowedOrigin($response, $request);
 
-        if ($this->options['maxAge'] !== null) {
-            $response->headers->set('Access-Control-Max-Age', $this->options['maxAge']);
-        }
+        $maxAge = $this->options['maxAge'];
 
-        $allowMethods = $this->options['allowedMethods'] === true
-            ? strtoupper($request->headers->get('Access-Control-Request-Method'))
-            : implode(', ', $this->options['allowedMethods']);
+        if ($this->options['allowedMethods'] === true) {
+            if ($this->options['supportsCredentials']) {
+                $allowMethods = strtoupper($request->headers->get('Access-Control-Request-Method'));
+                $this->varyHeader('Access-Control-Request-Method');
+                $maxAge = -1;
```

Comment:
Not sure if either of these are required, or that maxAge is implied to be different on different headers.

Reply:
It should follow the same logic as the main request I think. Basically make it static if if it can be, otherwise Vary it by the header being accessed.

Yes, in this case I think it's unnecessary to change the maxAge **and** add a Vary. The browser will know if the request header has changed (according to the `Vary`) and ignore the cache.